packages/js/settings-editor/src/components/custom-view/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import {
+	Fragment,
+	createElement,
+	useMemo,
+	useEffect,
+} from '@wordpress/element';
+import { sanitize } from 'dompurify';
+
+/**
+ * Sometimes extensions will place a <script /> tag in the custom output of a settings field,
+ * this function will extract the script content and return it as a SettingsField and the clean HTML.
+ *
+ * @param {string} html - The HTML content to process.
+ * @return {Object} An object containing the clean HTML and script settings.
+ */
+const processCustomView = ( html: string ) => {
+	const parser = new DOMParser();
+	const doc = parser.parseFromString( html, 'text/html' );
+
+	const scriptElements = Array.from(
+		doc.getElementsByTagName( 'script' )
+	).filter(
+		( script ) =>
+			script.getAttribute( 'type' ) === 'text/javascript' &&
+			script.textContent?.trim()
+	);
+
+	const scripts = scriptElements.map( ( script ) => {
+		const content = script.textContent as string;
+		script.remove();
+		return content;
+	} );
+
+	return {
+		cleanHTML: sanitize( doc.documentElement.outerHTML ),
+		scripts,
+	};
+};
+
+const SettingsScript = ( { content }: { content: string } ) => {
```

Comment:
I think most of the scripts need to be run after the elements are mounted. 

To ensure this is fully compatible with classic settings, they should be run in the order they appear in the document since these scripts are inline scripts in the HTML. However, I think that might be tricky and need more research.